core/src/main/java/org/geysermc/geyser/item/GeyserCustomItemOptions.java

```diff
@@ -31,6 +31,7 @@
 import java.util.OptionalInt;
 
 @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+@Deprecated
```

Comment:
I'd remove these - as long as api methods are marked as deprecated, we dont need to mark internal classes